### PR TITLE
Remove package-lock.json and clean up optional dependency handling

### PR DESCRIPTION
**Pull request by mykmartin**
Optional deps should now be installed just be adding `--install` to commands that require it (which just calls `npm install --no-save`). `sigh install`  is gone. `npm install/update` will *not* remove optional deps.

Travis will no longer build at the lock version set, but that wasn't really gaining us anything anyway. I've added a logging command to list the top-level packages and their versions in case that ever matters.

### CODE REVIEW

**Comment by mykmartin** [comment]
One consequence of this is that we lose the version info for the optional packages. I don't think we actually care that much about this, and if we end up needing a specific version (most likely due to forking something) we can just make sigh handle it.

File: .travis.yml [hunk 1]
```diff
@@ -21,8 +21,10 @@ script:
 cache: npm
 
 # Thinking about adding install steps? Why not modify the tools/setup script instead?
+# The list command dumps the top-level packages with their versions, in case we need that for debugging.
 install:
   - ./tools/setup
```
Comment (shaper) [inline .travis.yml]: Can you also update the setup script, see:

https://github.com/PolymerLabs/arcs/blob/master/tools/setup#L85

and L88

Comment (shaper) [reply .travis.yml]: Never mind... it's in here.

**Comment by shaper** [comment]
Sorry for delay, I just saw this. I support trying this out. Probably worth another PSA to team before you land it. I know you discussed it previously already.

**Comment by mykmartin** [comment]
No worries, I realise you weren't actually at work when I sent this :)

**Pull request merged**